src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -208,48 +198,41 @@ def update(self, power_status, error_codes):
         """
         Receives parts of a RobotStatus message
 
+        Saves the current time as the last robot status time
         Sets the robot UI as connected and updates the UI
-        Then sets the robot to disconnected with a timer callback to update the UI
+        Then sets a timer callback to disconnect the robot if needed
         :param power_status: The power status message for this robot
         :param error_codes: The error codes of this robot
         """
-        self.reconnect_robot()
+        self.time_of_last_robot_status = time.time()
 
-        self.update_ui(power_status, error_codes)
+        self.vision_pattern_label.setPixmap(self.color_vision_pattern)
 
-        self.mark_to_be_disconnected()
+        self.update_ui(power_status, error_codes)
 
-    def mark_to_be_disconnected(self):
-        """
-        Sets the boolean flag to disconnected
-        and calls a timer with a callback to grey out the UI in DISCONNECT_DURATION_MS
-        """
-        self.is_connected = False
-        if not self.to_be_disconnected:
-            QtCore.QTimer.singleShot(DISCONNECT_DURATION_MS, self.disconnect_robot)
-            self.to_be_disconnected = True
+        QtCore.QTimer.singleShot(DISCONNECT_DURATION_MS, self.disconnect_robot)
 
     def disconnect_robot(self):
         """
-        If robot is disconnected, attempts to get the lock
-        Once locked, if robot is still disconnected, greys out the robot UI
+        Calculates the time between the last robot status and now
+        If more than our threshold, resets UI
         """
-        if not self.is_connected:
-            self.connected_lock.acquire()
-            if not self.is_connected:
-                self.vision_pattern_label.setPixmap(self.bw_vision_pattern)
-                self.to_be_disconnected = False
-            self.connected_lock.release()
-
-    def reconnect_robot(self):
+        time_since_last_robot_status = time.time() - self.time_of_last_robot_status
+        if time_since_last_robot_status > DISCONNECT_DURATION_MS:
```

Comment:
There's a small bug making it not work right now.
```suggestion
        if time_since_last_robot_status > DISCONNECT_DURATION_MS * SECONDS_PER_MILLISECOND:
```